Extend pipelinerun to be able to select networkpolicy (#160)
SAP/stewardci-core: charts/steward/README.md

| <code>pipelineRuns.<wbr/>resourceQuota</code> | (string)<br/> The resource quota to be created in every pipeline run namespace. The value must be a string containing a complete `resourcequotas` resource manifest in YAML format. The `.metadata` section of the manifest can be omitted, as it will be replaced anyway. See the [Kubernetes documentation of resource quotas][k8s-resourcequotas] for details about Kubernetes resource quotas.| none |

### Feature Flags

| Parameter | Description | Default |
|---|---|---|
| <code>featureFlags</code> | (string)<br/> Feature flag definition.  | empty |

> The feature flags definition is a string with one or more feature flags. Multiple feature flags are separated by komma `,` or space `' '`.
> A feature flag can be set to enabled with adding the name optionally prefixed with a plus `+`.
> A feature flag can be set to disabled by adding the name prefixed with a dash `-`.

Reviewer: ```suggestion
The feature flags definition is a string containing any number of feature flag names separated by any non-empty sequence of comma (`,`) and/or whitespace (space, horizontal tab, vertical tab, carriage return, newline, form feed). Separators at the beginning and the end of the string are allowed.

A feature flag gets _enabled_ if the name is either prefixed with a plus sign (`+`) or not disabled.
A feature flag gets _disabled_ if the name is prefixed with a minus sign (`-`).

**Example 1: Typical format**

    Flag1 +Flag2 -Flag3

or alternatively using comma and space as separator:

    Flag1, +Flag2, -Flag3

or alternatively using newline as separator:

    Flag1
    +Flag2
    -Flag3

`Flag1` and `Flag2` get enabled, while `Flag3` gets disabled.

**Example 2: Untypical but correct format**

    , Flag1 +Flag2, ,,,,
    Flag3,

It has the same effect as in example 1. The definition string has leading and trailing separators, uses different separator sequences.
```

| Feature Flag | Description | Default |
| --- | --- | --- |

> | RetryOnInvalidPipelineRunsConfig | If enabled: Retries a pipeline run on a configuration error until the configuration error if fixed or the timeout is reached.</br>If disabled: The pipeline run is finished with an infrastructure error. | disabled |

Reviewer: ```suggestion
| RetryOnInvalidPipelineRunsConfig | If enabled, the pipeline run controller retries reconciling PipelineRun objects in case the controller configuration (in ConfigMaps) is invalid or cannot be loaded. It is assumed that the condition can be detected by a monitoring tool, triggers an alert and operators fix the issue in a timely manner. By that operator errors do not immediately break user pipeline runs. However, processing of PipelineRun objects may be delayed significantly in case of invalid configuration.<br/><br/> If disabled, the current behavior is used: immediately set all unfinished PipelineRun objects to finished with result code `error_infra`.<br/><br/>  The new behavior is supposed to become the default in a future release of Steward. | disabled |
```

| <code>pipelineRuns.<wbr/>resourceQuota</code> | (string)<br/> The resource quota to be created in every pipeline run namespace. The value must be a string containing a complete `resourcequotas` resource manifest in YAML format. The `.metadata` section of the manifest can be omitted, as it will be replaced anyway. See the [Kubernetes documentation of resource quotas][k8s-resourcequotas] for details about Kubernetes resource quotas.| none |

### Feature Flags

| Parameter | Description | Default |
|---|---|---|
| <code>featureFlags</code> | (string)<br/> Feature flag definition.  | empty |

The feature flags definition is a string containing any number of feature flag names separated by any non-empty sequence of comma (`,`) and/or whitespace (space, horizontal tab, vertical tab, carriage return, newline, form feed). Separators at the beginning and the end of the string are allowed.

> A feature flag gets _enabled_ if the name is either prefixed with a plus sign (`+`) or not disabled.

Author: ```suggestion
A feature flag gets _enabled_ if the name is either prefixed with a plus sign (`+`) or not prefixed.
```
This may be misleading. If the feature flag default is set to false and the flag is missing it is not enabled.

Reviewer: > This may be misleading. If the feature flag default is set to false and the flag is missing it is not enabled.

Then I'd prefer this:
```suggestion
A feature flag gets _enabled_ if the name is either prefixed with a plus sign (`+`) or not prefixed with a minus sign (`-`).
```

A feature flag gets _disabled_ if the name is prefixed with a minus sign (`-`).

**Example 1: Typical format**

    Flag1 +Flag2 -Flag3

or alternatively using comma and space as separator:

    Flag1, +Flag2, -Flag3

or alternatively using newline as separator:

    Flag1
    +Flag2
    -Flag3

`Flag1` and `Flag2` get enabled, while `Flag3` gets disabled.

**Example 2: Untypical but correct format**

    , Flag1 +Flag2, ,,,,
    Flag3,

> It has the same effect as in example 1. The definition string has leading and trailing separators, uses different separator sequences.

Reviewer: ```suggestion
It has the same effect as in example 1.
The definition string has leading and trailing separators and uses different separator sequences.
```

> | Feature Flag | Description | Default |
> | --- | --- | --- |
> | RetryOnInvalidPipelineRunsConfig | If enabled, the pipeline run controller retries reconciling PipelineRun objects in case the controller configuration (in ConfigMaps) is invalid or cannot be loaded. It is assumed that the condition can be detected by a monitoring tool, triggers an alert and operators fix the issue in a timely manner. By that operator errors do not immediately break user pipeline runs. However, processing of PipelineRun objects may be delayed significantly in case of invalid configuration.<br/><br/> If disabled, the current behavior is used: immediately set all unfinished PipelineRun objects to finished with result code `error_infra`.<br/><br/>  The new behavior is supposed to become the default in a future release of Steward. | disabled |

Reviewer: ```suggestion

#### List of Defined Feature Flags

| Feature Flag | Description | Default |
| --- | --- | --- |
| `RetryOnInvalidPipelineRunsConfig` | If enabled, the pipeline run controller retries reconciling PipelineRun objects in case the controller configuration (in ConfigMaps) is invalid or cannot be loaded. It is assumed that the condition can be detected by a monitoring tool, triggers an alert and operators fix the issue in a timely manner. By that operator errors do not immediately break user pipeline runs. However, processing of PipelineRun objects may be delayed significantly in case of invalid configuration.<br/><br/> If disabled, the current behavior is used: immediately set all unfinished PipelineRun objects to finished with result code `error_infra`.<br/><br/>  The new behavior is supposed to become the default in a future release of Steward. | disabled |
```
